src/audio_worklet.js

```diff
@@ -47,79 +47,89 @@ function createWasmAudioWorkletProcessor(audioParams) {
       // then perform the wasm audio worklet call,
       // and finally marshal audio output data back.
 
-      let numInputs = inputList.length,
-        numOutputs = outputList.length,
-        numParams = 0, i, j, k, dataPtr,
-        bytesPerChannel = this.samplesPerChannel * 4,
-        stackMemoryNeeded = (numInputs + numOutputs) * {{{ C_STRUCTS.AudioSampleFrame.__size__ }}},
-        oldStackPtr = stackSave(),
-        inputsPtr, outputsPtr, outputDataPtr, paramsPtr,
-        didProduceAudio, paramArray;
+      var numInputs = inputList.length;
+      var numOutputs = outputList.length;
+      var numParams = 0;
+
+      var entry; // reused list entry or index
+      var subentry; // reused channel or other array in each list entry or index
+      var structPtr; // working start to each struct record
+      var dataPtr; // start of the data section, usually after all structs
+
+      var bytesPerChannel = this.samplesPerChannel * {{{ getNativeTypeSize('float') }}};
+      var stackMemoryNeeded = (numInputs + numOutputs) * {{{ C_STRUCTS.AudioSampleFrame.__size__ }}};
+      var inputsPtr, outputsPtr, outputDataPtr, paramsPtr; // pointers passed to callback()
+
+      var oldStackPtr = stackSave();
 
       // Calculate how much stack space is needed.
-      for (i of inputList) stackMemoryNeeded += i.length * bytesPerChannel;
-      for (i of outputList) stackMemoryNeeded += i.length * bytesPerChannel;
-      for (i in parameters) stackMemoryNeeded += parameters[i].byteLength + {{{ C_STRUCTS.AudioParamFrame.__size__ }}}, ++numParams;
+      for (entry of inputList) stackMemoryNeeded += entry.length * bytesPerChannel;
+      for (entry of outputList) stackMemoryNeeded += entry.length * bytesPerChannel;
+      for (entry in parameters) {
+        stackMemoryNeeded += parameters[entry].byteLength + {{{ C_STRUCTS.AudioParamFrame.__size__ }}};
+        ++numParams;
+      }
 
       // Allocate the necessary stack space.
       inputsPtr = stackAlloc(stackMemoryNeeded);
 
       // Copy input audio descriptor structs and data to Wasm
-      k = inputsPtr;
+      structPtr = inputsPtr;
       dataPtr = inputsPtr + numInputs * {{{ C_STRUCTS.AudioSampleFrame.__size__ }}};
```

Comment:
Maybe just declare there down here.. no need to declare them above, right?

Reply:
I believe in the old code the `var`s were all group together to avoid repeating the `var` keyword.   Now that we can run closure on this code we should just declare each `var` at the place where its first needed.

Reply:
Quick check, if I move them around to be near their use site it's a delta of `6`. I think it's worth it, your call.

Reply:
That makes no sense to me since I though closure took care of that.. oh well.  lgtm.